### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by AbrahamFergie**

### CODE REVIEW

**Comment by AbrahamFergie** [comment]
Would like feedback on if the pseudo code in the hashFunction file is indeed a correct format for how a hashtable should look.

File: spec/hashTable.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+describe('hashTable', () => {
+  it('exists', () => {
+    expect(HashTable).to.be.a('function')
+  })
+  context('put()', () => {
+    it.only('adds a key-value pair to the hash table.', () => {
+      const ht = new HashTable()
+      ht.put('name', 'sal')
+      console.log('where are you', ht);
+      expect(ht.dataStore.name).to.eql('sal')
```
Comment (jrobcodes) [inline spec/hashTable.js]: Consider using only the public API of the `HashTable` to test. This is dependent on the underlying representation of data in your `HashTable`, so if the underlying representation changes - even though the implementation still works - all of your tests will fail.

**Review by jrobcodes — COMMENTED** [review]
If I understand the request correctly, no, the pseudo code does not look like how I would typically implement a `HashTable`.  A `HashTable` usually stores a statically sized array (which doesn't exist in Javascript, but we can act as if it does), where each index of the array holds a reference to a Linked List.  The hashFunction maps the key space to the size of the array, and insertions happen into the Linked Lists at the index obtained from the hashFunction.

File: spec/hashTable.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    it('returns true if the hash table contains the key.', () => {
+      const ht = new HashTable()
+      expect(ht.contains('name')).to.be.false
+      ht.put('name', 'frankenstein')
+      ht.put('19.0', 'stein')
+      expect(ht.contains('name')).to.be.true
+    })
+  })
+  context('remove()', () => {
+    it('removes a key-value pair by key.', () => {
```
Comment (EthanJStark) [inline spec/hashTable.js]: I'm sure you're on your way to this, but:
```js
  context('remove()', () => {
    it('removes a key-value pair by key.', () => {
      const ht = new HashTable()
      ht.put('name', 'frankenstein')
      expect(ht.contains('name')).to.be.true
      ht.remove('name')
      expect(ht.contains('name')).to.be.false
    })
  })
```

And then test that size decreases.

**Review by EthanJStark — COMMENTED** [review]
Abraham! Your hash-table is looking great! Very readable, and seems to be functioning properly.  Couldn't find much of anything to improve, just jotted down an idea on completing the `remove()` spec.

**Pull request merged**